tests/system-tests/rdscore/internal/rdscorecommon/egress-ip.go

```diff
@@ -0,0 +1,694 @@
+
+	return deployContainer
+}
+
+func sendTrafficCheckIP(clientPods []*pod.Builder, isIPv6 bool, expectedIPs []string) error {
+	By("Validating pods source address")
+
+	targetIP := RDSCoreConfig.EgressIPRemoteIP
+
+	if isIPv6 {
+		targetIP = fmt.Sprintf("[%s]", RDSCoreConfig.EgressIPRemoteIPv6)
+	}
+
+	cmdToRun := []string{"/bin/bash", "-c",
+		fmt.Sprintf("curl --connect-timeout 5 -Ls http://%s:%s/clientip",
+			targetIP, RDSCoreConfig.EgressIPTcpPort)}
+
+	for _, clientPod := range clientPods {
+		var parsedIP string
+
+		err := wait.PollUntilContextTimeout(
+			context.TODO(),
+			time.Second,
+			time.Second*5,
+			true,
+			func(ctx context.Context) (bool, error) {
+				result, err := clientPod.ExecCommand(cmdToRun, clientPod.Object.Spec.Containers[0].Name)
+
+				if err != nil {
+					glog.V(100).Infof("Error running command from within a pod %q: %v",
+						clientPod.Object.Name, err)
+
+					return false, nil
+				}
+
+				glog.V(100).Infof("Successfully executed command from within a pod %q: %v",
+					clientPod.Object.Name, err)
+				glog.V(100).Infof("Command's output:\n\t%v", result.String())
+
+				parsedIP, _, err = net.SplitHostPort(result.String())
+
+				if err != nil {
+					glog.V(100).Infof("Failed to parse %q for host/port pair", result.String())
+
+					return false, nil
+				}
+
+				glog.V(100).Infof("Verify IP version type correctness")
+
+				myIP, err := netip.ParseAddr(parsedIP)
+
+				if err != nil {
+					glog.V(100).Infof("Failed to parse used ip address %q", parsedIP)
+
+					return false, nil
+				}
+
+				if isIPv6 && myIP.Is4() || !isIPv6 && myIP.Is6() {
+					glog.V(100).Infof("Wrong IP version detected; %q", parsedIP)
+
+					return false, nil
+				}
+
+				glog.V(100).Infof("Comparing %q with expected %q", parsedIP, expectedIPs)
+
+				for _, expectedIP := range expectedIPs {
+					if parsedIP == expectedIP {
+						return true, nil
+					}
+				}
+
+				glog.V(100).Infof("Mismatched IP address. Expected %q got %q", expectedIPs, parsedIP)
+
+				return false, nil
+			})
+
+		if err != nil {
+			return fmt.Errorf("failed to run command from within pod %s: %w", clientPod.Object.Name, err)
+		}
+	}
+
+	return nil
+}
+
+func getExpectedEgressIPList() ([]string, error) {
+	By("Getting a map of source nodes and assigned Egress IPs for these nodes")
+
+	egressIPObj, err := egressip.Pull(APIClient, RDSCoreConfig.EgressIPName)
+
+	if err != nil {
+		glog.V(100).Infof("Failed to pull egressIP %q object: %v", RDSCoreConfig.EgressIPName, err)
+
+		return nil, fmt.Errorf("failed to retrieve egressIP %s object: %w", RDSCoreConfig.EgressIPName, err)
+	}
+
+	egressIPSet, err := egressIPObj.GetAssignedEgressIPMap()
+
+	if err != nil {
+		glog.V(100).Infof("Failed to retrieve egressIP %s assigned egressIPs map: %v",
+			RDSCoreConfig.EgressIPName, err)
+
+		return nil, fmt.Errorf("failed to retrieve egressIP %s assigned egressIPs map: %w",
+			RDSCoreConfig.EgressIPName, err)
+	}
+
+	if len(egressIPSet) != 2 {
```

Comment:
is this a _strict requirement_ from the test? can we tolerate if there are more or less IPs and fail only if there's *0* detected?

Reply:
our test is based on the fact that we use two workers assigned to the egressIP service, and we will check both (test prerequisite): ipv4 and ipv6 => it has to be exactly two records found, otherwise -> fail the test